docs/serialization.rst

```diff
@@ -0,0 +1,176 @@
+Hazelcast Serialization
+=======================
+
+Hazelcast Python Client is a binary compatible implementation of `Hazelcast serialization <http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#serialization>`_.
+
+Content:
+
+- Builtin serializers
+- :class:`~hazelcast.serialization.api.IdentifiedDataSerializable`
+- :class:`~hazelcast.serialization.api.Portable`
+- Custom Serializers
+- Global serializers
+
+**Warning :**
+Python client do not support Dataserializable as it's a java class name dependency. Please use :class:`~hazelcast.serialization.api.IdentifiedDataSerializable` instead.
+
+
+Builtin serializers
+-------------------
+
+For python builtin types; ``None``, ``bool``, ``int``, ``long``, ``float``, ``str``, arrays of them are supported by default.
+
+Integer Type: Python has a different int type than Java. Python 2.x has integer type based on the data size which is int or long.
+So in python we have a configuration for int size to map server data type.
+
+You can configure the default integer type as :
+
+.. code-block:: python
+
+    config.serialization_config.default_integer_type = INTEGER_TYPE.INT
+
+Please see ::const:`~hazelcast.config.INTEGER_TYPE` for details. Please be careful with INTEGER_TYPE.VAR as static type lang's cannot deserialize it.
+
+There exist an internal serializer for any custom class that is not registered a serializer such as IdentifiedDataSerializable, Portable or Custom Serializer.
+So the serializer service search for a serializer for a custom class, if it fails to find one it uses python builtin serialization :mod:`cPickle` to handle it.
```

Comment:
the serializer service searches for a serializer for a custom class; if it fails to find one, it uses Python...